### PR TITLE
[BENCHMARK] Lazy ChildContainer with full termination sequence

### PR DESCRIPTION
**Pull request by Aaronontheweb**
Benchmarking #1772 in combination with #1765

### CODE REVIEW

File: src/core/Akka/Actor/ChildrenContainer/Internal/ChildrenContainerBase.cs [hunk 1]
```diff
@@ -30,25 +54,29 @@ protected ChildrenContainerBase(IImmutableDictionary<string, IChildStats> childr
...
-        public IReadOnlyList<IInternalActorRef> Children
+        public IReadOnlyCollection<IInternalActorRef> Children
         {
             get
             {
-                return (from stat in InternalChildren.Values
-                        let childRestartStats = stat as ChildRestartStats
-                        where childRestartStats != null
-                        select childRestartStats.Child).ToList();
+                var children = from stat in InternalChildren.Values
```
Comment (annymsMthd) [inline src/core/Akka/Actor/ChildrenContainer/Internal/ChildrenContainerBase.cs]: it's kinda a personal thing but I find this linq syntax a little meh.
can we just do something like
`
return InternalChildren.Values
                    .OfType<ChildRestartStats>()
                    .Select(stats => stats.Child)
`

File: src/core/Akka/Actor/ChildrenContainer/Internal/ChildrenContainerBase.cs [hunk 2]
```diff
@@ -30,25 +54,29 @@ protected ChildrenContainerBase(IImmutableDictionary<string, IChildStats> childr
...
             }
         }
 
-        public IReadOnlyList<ChildRestartStats> Stats
+        public IReadOnlyCollection<ChildRestartStats> Stats
         {
             get
             {
-                return (from stat in InternalChildren.Values
+                var children = from stat in InternalChildren.Values
```
Comment (annymsMthd) [inline src/core/Akka/Actor/ChildrenContainer/Internal/ChildrenContainerBase.cs]: same here. I find:
`return InternalChildren.Values.OfType<ChildRestartStats>();`
a little easier to read

File: src/core/Akka.Tests/Actor/ActorSystemSpec.cs [hunk 3]
```diff
@@ -185,14 +188,15 @@ public void Reliable_deny_creation_of_actors_while_shutting_down()
                 }
 
                 
-                if (!failing && sys.Uptime.TotalSeconds >= 5)
+                if (!failing && sys.Uptime.TotalSeconds >= 10)
                     throw new AssertionFailedException(created.Last() + Environment.NewLine +
                                                        "System didn't terminate within 5 seconds");
```
Comment (Silv3rcircl3) [inline src/core/Akka.Tests/Actor/ActorSystemSpec.cs]: The failure message still says 5 seconds ;)

**Comment by JeffCyr** [comment]
I updated PR #1772 to fix the Children.Count that might not have been accurate.

I also changed the LINQ queries to make @annymsMthd happy :)

File: src/core/Akka.Tests/Actor/ActorSystemSpec.cs (hunk 3 above)
Comment (Aaronontheweb) [inline src/core/Akka.Tests/Actor/ActorSystemSpec.cs]: Yeah, it's that way on the JVM too. Figured they were just being generous :p

**Pull request closed without merging**